src/library/socket.cpp

```diff
@@ -210,7 +210,7 @@ Address UdpSocket::GetPeerAddr() const
     ASSERT(getpeername(mNetCtx.fd, reinterpret_cast<sockaddr *>(&addr), &len) == 0);
 
     Address ret;
-    ASSERT(ret.Set(addr) == Error::kNone);
+    ASSERT(ret.Set(addr).NoError());
```

Comment:
`ASSERT` in OT Commissioner has its implementation that it is not removed in release build so it can have side effect, this is why it is defined.

I am okay to change the behavior of `ASSERT` to follow `assert` and avoid side effect.